lib/utils/stream_extension.dart

```diff
@@ -45,4 +47,44 @@ extension StreamExtension on Stream {
     };
     return controller.stream;
   }
+
+  Stream<SyncUpdate> rateLimitWithSyncUpdate(Duration t) {
+    final controller = StreamController<SyncUpdate>();
+    Timer? timer;
+    SyncUpdate? pendingMessage;
+
+    void processMessage() {
```

Comment:
please move this method outside of `rateLimitWithSyncUpdate` to improve readability